[Rehad A] iP

xlunarin opened the pull request:
### Duke 

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html))

Duke helps you in managing and organizing your tasks. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/rehad-a/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you!

And it's **FREE**! 🥳 🎉 

Features:

- [x] Managing tasks 📝 
- [ ] Managing deadlines (coming soon) 📅 
- [ ] Reminders (coming soon) 🔔 

--------------
If you're a Java programmer, you can use it to practice Java too. Here's `main` method:
```java
public class Main {
    public static void main(String[] args[]) {
        Application.laungh(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

xlunarin committed:
implemented Level-1

xlunarin committed:
Implemented Level-2

xlunarin committed:
implemented Level-3 and added Task.java

xlunarin committed:
implemented Level-4. Added DeadlineTask.java, EventTask.java, and TodoTask.java

xlunarin committed:
updated input.txt and EXPECTED.txt

xlunarin committed:
implemented Level-5 and added DukeException.java

xlunarin committed:
implemneted Level-6

xlunarin committed:
Implement level 7

xlunarin committed:
Implement level-7

xlunarin committed:
Merge commit 'b87bebf49b9ed27ed89468dc9d9f35904eb40f07' into branch-Level-7

xlunarin committed:
Implement Level-8

xlunarin committed:
Merge branch 'branch-Level-8'

simonjulianl commented on .gitignore:
```diff
@@ -15,3 +15,8 @@ bin/
+src/main/java/DeadlineTask.class
+src/main/java/Duke.class
+src/main/java/EventTask.class
+src/main/java/Task.class
```
Should this be simplified as `src/main/java/*.class` to make the file less verbose ?

simonjulianl commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,21 @@
+
+    public DeadlineTask(String description, String time) {
+        super(description);
+        this.type = "D";
```
Should we remove `this` as these variables are not shadowed

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
 public class Duke {
-    public static void main(String[] args) {
+    public static void loadTasksFromFile(ArrayList<Task> tasks) throws IOException, DukeException {
+        String directoryPath = "./data";
```
Should we add visibility modifier such as `private` for these variables

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+import jdk.jfr.Event;
```
Is this unused imports ?

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+        Scanner f = new Scanner(dataFile);
+
+        while (f.hasNext()) {
+            String[] task = f.nextLine().split("|");
```
Should we use plural form for `task`

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                        System.out.printf("%d.%s\n", i + 1, tasks.get(i));
+                    }
+                    System.out.println(line);
+                }else if (command.contains("delete")) {
```
Should we fix the spacing here ?

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                    int taskIdxStart = command.indexOf(" ") + 1;
+                    String task = command.substring(taskIdxStart);
+                    tasks.add(new TodoTask(task));
+                    System.out.printf("%sGot it. I've added this task:\n%s\nNow you have %d tasks in the list\n%s\n", line, tasks.get(ctr), ctr + 1, line);
```
Should we separate this into 2 lines, as 1 line has max of 120 chars

simonjulianl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+    }
+
+    public static void saveTasksToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter("./data/duke.txt");
```
Really nice use of file writer btw! just one suggestion you can check `try-with-resources`

simonjulianl reviewed (changes requested):
Overall pretty clean code and functional! Just maybe more spacing and newlines so that people can read it easier

xlunarin committed:
Implement Level-9

xlunarin committed:
Make the code more OOP

xlunarin committed:
Organize into packages

xlunarin committed:
Add JUnit test

xlunarin committed:
Create a JAR file

xlunarin committed:
Add JavaDoc

xlunarin committed:
Follow the coding standard

xlunarin committed:
Remove thrown exception from main header

xlunarin committed:
Merge branch 'add-gradle-support'

xlunarin committed:
Use Gradle

xlunarin committed:
Use CheckStyle

xlunarin committed:
Merge branch 'branch-A-CodingStandard'

yyhangz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                if (command.contains("done")) {
+                    if (!command.contains(" ")) {
+                        throw new DukeException("OOPS!! done needs the index of the task.");
+                    }
```
Should we use an alternative implementation that reduces nesting?

yyhangz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+        System.out.println(line + "Hello I'm Duke\nWhat can I do for you?\n" + line);
+
+        ArrayList<Task> tasks = new ArrayList<>(100);
+        int ctr = 0;
```
Should we use a name that is more semantically appropriate for the variable?

yyhangz commented on src/main/java/EventTask.java:
```diff
@@ -0,0 +1,20 @@
+
+    public EventTask(String description, boolean isDone, String time) {
+        super(description, isDone);
+        this.type = "E";
```
Seems like there is repetition in the assignment of `this.type = "E"`. Perhaps this could be abstracted for ease of refactoring in the future?

yyhangz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+            String taskType = task[0];
+            boolean isDone = (Integer.parseInt(task[1]) == 1);
+            String description = task[2];
+            String time = task.length == 4 ? task[3] : null;
```
Perhaps `4` could be defined as a class level constant to avoid the use of magic literals?

yyhangz reviewed (commented):
Great implementation overall!

xlunarin committed:
Implement GUI

xlunarin committed:
Edit README.md

xlunarin committed:
Implement GUI

xlunarin committed:
Application does not close when calling a bye command

Duke only returns a bye message to the user without exiting the
application.

Let's update Duke to exit the application after sending the bye message

xlunarin committed:
Send a greeting message upon launching Duke

xlunarin committed:
Add assertions

xlunarin committed:
Improve code quality

xlunarin committed:
Merge pull request #2 from rehad-a/branch-Level-A-Assertions

Add assertions

xlunarin committed:
Merge branch 'master' into branch-Level-A-CodeQuality

xlunarin committed:
Merge branch 'branch-Level-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Storage.java

xlunarin committed:
Add update command

xlunarin committed:
Merge commit '0474acd7f6de3ca6442499840062f6618ef9fbfa'

# Conflicts:
#	src/main/java/duke/Storage.java

xlunarin committed:
Better GUI

xlunarin committed:
Enhance GUI

xlunarin committed:
Set theme jekyll-theme-minimal

xlunarin committed:
Set theme jekyll-theme-minimal

xlunarin committed:
Improve GUI

xlunarin committed:
User guide

xlunarin committed:
Merge branch 'master' of https://github.com/rehad-a/ip

xlunarin committed:
Set theme jekyll-theme-minimal

xlunarin committed:
Fix README.md

xlunarin committed:
Merge branch 'master' of https://github.com/rehad-a/ip

xlunarin committed:
Delete _config.yml

xlunarin committed:
Update README.md

xlunarin committed:
Update README.md
